### PR TITLE
Feature/1177/missing documentation

### PR DESCRIPTION
**Pull request by garyluu**
Missing a few sections of documents
Completely missing the "For developers document" (was replaced by an identically copy of the docker-registries document)

Edit to add: making #1177 linked

### CODE REVIEW

**Comment by agduncan94** [comment]
Looks like you removed some of the existing documentation.

**Comment by garyluu** [comment]
It's still here at https://github.com/ga4gh/dockstore/blob/feature/1177/missingDocumentation/_docs/docker-registries.md

File: _docs/for-developers.md [hunk 1]
```diff
@@ -2,65 +2,172 @@
...
+Downloading: #input_file from s3://oicr.temp/bamstats_report.zip into directory: /media/large_volume/dockstore_tools/dockstore-tool-md5sum/./datastore/launcher-a246f1b6-21fd-468e-8780-b064d311dda5/inputs/73b70f11
+-1711-40b7-bfea-9ee4543a8226
+Found file s3://oicr.temp/bamstats_report.zip in cache, hard-linking
+Calling on plugin io.dockstore.provision.S3Plugin$S3Provision to provision s3://oicr.temp/bamstats_report.zip
+Calling out to cwltool to run your tool
+Executing: cwltool --enable-dev --non-strict --outdir /media/large_volume/dockstore_tools/dockstore-tool-md5sum/./datastore/launcher-a246f1b6-21fd-468e-8780-b064d311dda5/outputs/ --tmpdir-prefix /media/large_volu
+me/dockstore_tools/dockstore-tool-md5sum/./datastore/launcher-a246f1b6-21fd-468e-8780-b064d311dda5/tmp/ --tmp-outdir-prefix /media/large_volume/dockstore_tools/dockstore-tool-md5sum/./datastore/launcher-a246f1b6-
+21fd-468e-8780-b064d311dda5/working/ /tmp/1488407859906-0/temp3047430238970788171.cwl /media/large_volume/dockstore_tools/dockstore-tool-md5sum/./datastore/launcher-a246f1b6-21fd-468e-8780-b064d311dda5/workflow_p
+arams.json
+/usr/local/bin/cwltool 1.0.20170217172322
```
Comment (denis-yuen) [inline _docs/for-developers.md]: @garyluu @agduncan94 
Note: not part of this PR but let's try to get Jenkins running this weekend, we might want to pick a newer cwltool to test against too
